### DISCUSSION TITLE
Is the unsubscribeAll method of the Delegate incorrectly written?

### DISCUSSION THREAD

**Discussion by zaleGZL** [tradingview/lightweight-charts#994] [General] [answered]
https://github.com/tradingview/lightweight-charts/blob/1d49f27ce7aa130ae0579dfdbb4ee2b7494c6c4e/src/helpers/delegate.ts#L30

should listener.linkedObject !== linkedObject  ?

**Comment by timocov** [comment, marked as answer]
Nice catch @zaleGZL! Of course it should be as you described. I just created an issue to fix this #995. Do you want to provide a fix for that (or cover it with tests)?

**Reply by zaleGZL** [reply]
Sure, I submitted a PR.
